### PR TITLE
 Remove unused activation from hipblaslt asm kernel

### PR DESCRIPTION
**Pull request by StaceyLai**
The ASM kernel supports 10 activation functions. However, the hipBLASLT API only supports 3 of these: Gelu, Relu, and DGelu.
This means that the ASM kernel in hipBLASLT contains redundant activation code that is never used.

This PR removes the unused activation functions from hipBLASLT. (The TensileLite part still retains all 10 activation functions.)
ActivationType supports only 'all' and 'hipblaslt_all' currently. It is active only when the Activation configuration is set to True.
Otherwise, ActivationType will be set to 'none'.

### CODE REVIEW

**Comment by StaceyLai** [comment]
There are too many files changed in the last commit, which make its diff not be able to display on GitHub.
I'll separate it.

**Comment by StaceyLai** [comment]
There are too many files changed in the last commit, which make its diff not be able to display on GitHub. Because I add "ActivationType: hipblaslt_all" in the all logic YAML.

File: tensilelite/Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp [hunk 1]
```diff
@@ -1981,7 +1981,9 @@ namespace Tensile
 
                 virtual bool operator()(ContractionProblemGemm const& problem) const override
                 {
-                    return problem.activationType() == value || value == ActivationType::All;
+                    return problem.activationType() == value
+                           || value == ActivationType::All
+                           || value == ActivationType::Hipblaslt_all;
```
Comment (jichangjichang) [inline tensilelite/Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp]: for Hipblaslt_all, need to check if the problem's activation is in hipblaslt act supported list.

File: tensilelite/Tensile/SolutionStructs.py [hunk 2]
```diff
@@ -531,7 +535,7 @@ def __str__(self):
 
     # Activation Naming
     if self["ActivationType"] != 'none':
-      if self["ActivationType"] == 'all':
+      if self["ActivationType"] in ['all', 'hipblaslt_all']:
         name += "_A"
```
Comment (jichangjichang) [inline tensilelite/Tensile/SolutionStructs.py:539]: all and hipblaslt_all should have different ShortName

Comment (StaceyLai) [reply tensilelite/Tensile/SolutionStructs.py:539]: Fixed, please help to review.

File: tensilelite/Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp (hunk 1 above)
Comment (StaceyLai) [reply tensilelite/Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp]: Fixed in the new patch set. Please help to review.

Comment (StaceyLai) [reply tensilelite/Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp]: The new patch set version will cause a lot of failure:
Failed
NO solution found!

Further discussion is needed to determine the correct solution.

**Comment by StaceyLai** [comment]
Library size:
Before: 13038M (codebase: commit 65e956b7075a854229c308cd4119ddaaec287122)
After: 12573M

**Review by jichangjichang — APPROVED** [review]
LGTM

**Comment by StaceyLai** [comment]
Pass tox testing:
"
---- generated xml file: /stacelai_5/hipBLASLt/tensilelite/python_tests.xml ----
[33m========= [32m75 passed[0m, [33m[1m31 skipped[0m, [33m[1m1235 warnings[0m[33m in 11075.90s (3:04:35)[0m[33m ==========[0m
py3: exit 0 (11076.18 seconds) /stacelai_5/hipBLASLt/tensilelite> py.test -v --basetemp=/stacelai_5/hipBLASLt/tensilelite/CI_out/py3/tmp --junit-xml=/stacelai_5/hipBLASLt/tensilelite/python_tests.xml --junit-prefix=py3 --color=yes -n 4 --prebuilt-client=/stacelai_5/hipBLASLt/tensilelite/CI_out/py3/client/0_Build/client/tensile_client Tensile/Tests -m common pid=3261565
  py3: OK (11206.85=setup[8.05]+cmd[0.42,122.19,11076.18] seconds)
  congratulations :) (11206.89 seconds)
"

Pass hipblaslt-test:
"
[----------] 1 test from ExtOpTest/ExtOpAMaxWithScaleUnsupportedDatatypeTest
[ RUN      ] ExtOpTest/ExtOpAMaxWithScaleUnsupportedDatatypeTest.amaxWithScaleFailureUnsupportedDatatype/0
[       OK ] ExtOpTest/ExtOpAMaxWithScaleUnsupportedDatatypeTest.amaxWithScaleFailureUnsupportedDatatype/0 (0 ms)
[----------] 1 test from ExtOpTest/ExtOpAMaxWithScaleUnsupportedDatatypeTest (0 ms total)

[----------] Global test environment tear-down
[==========] 48206 tests from 13 test suites ran. (2175522 ms total)
[  PASSED  ] 48206 tests.
hipBLASLt version: 1000

command line: ./build/release/clients/staging/hipblaslt-test 
"

**Review by Jinp800125 — APPROVED** [review]

**Pull request merged**